packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -159,6 +177,22 @@ function renderHealthCheck(hc: HealthCheck | undefined, pm: PortMapping): CfnVir
     timeoutMillis: (hc.timeout || cdk.Duration.seconds(2)).toMilliseconds(),
     unhealthyThreshold: hc.unhealthyThreshold || 2,
   };
+
+  (Object.keys(healthCheck) as Array<keyof CfnVirtualNode.HealthCheckProperty>)
+    .filter((key) => HEALTH_CHECK_PROPERTY_THRESHOLDS[key] && typeof healthCheck[key] === 'number')
```

Comment:
You probably also want to filter for `!Token.isUnresolved`.

Reply:
Wouldn't `typeof healthCheck[key] === 'number'` cover it?

Reply:
I guess `healthCheck` could be unresolved, but I'd have to check it earlier

Reply:
No, Tokens can also be encoded into numbers. It will look like `-1.358238248E-18`.

Reply:
Got it, thanks. Done